gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/HadoopFsEndPointDataset.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.replication;
+
+import java.io.IOException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import gobblin.dataset.FileSystemDataset;
+import gobblin.util.HadoopUtils;
+
+
+public class HadoopFsEndPointDataset implements FileSystemDataset{
```

Comment:
Tried to use the VersionFinder to select valid sub directory to replicate. The AbstractDatasetVersionFinder.findDatasetVersions function take Dataset as input. So I created this class as the Dataset wrapper for HadoopFsEndPoint